src/SqlBindingUtilities.cs

```diff
@@ -290,5 +292,73 @@ internal static bool IsFatalSqlException(this Exception e)
             }
             return true;
         }
+
+        /// <summary>
+        /// Get the Server Properties for the given connection.
+        /// </summary>
+        /// <returns>ServerProperties (EngineEdition and Edition) of the target Sql Server.</returns>
+        public static async Task<ServerProperties> GetServerTelemetryProperties(SqlConnection connection, ILogger logger, CancellationToken cancellationToken)
+        {
+            if (TelemetryInstance.Enabled)
+            {
+                string serverPropertiesQuery = $"SELECT SERVERPROPERTY('EngineEdition'), SERVERPROPERTY('EngineEdition')";
+
+                logger.LogDebugWithThreadId($"BEGIN GetServerTelemetryProperties Query={serverPropertiesQuery}");
+                using (var selectServerEditionCommand = new SqlCommand(serverPropertiesQuery, connection))
+                using (SqlDataReader reader = await selectServerEditionCommand.ExecuteReaderAsync(cancellationToken))
+                {
+                    try
```

Comment:
Try try/catch should go around everything starting right after the if - the main thing being to catch possible exceptions from ExecuteReaderAsync

Reply:
I thought I added it after if, my bad. Thanks for catch :)